packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -1758,15 +1681,18 @@ export class Cluster extends ClusterBase {
     // and configured to allow connections from itself.
     this.kubectlSecurityGroup = this.clusterSecurityGroup;
 
-    this.adminRole.assumeRolePolicy?.addStatements(new iam.PolicyStatement({
-      actions: ['sts:AssumeRole'],
-      principals: [this.kubectlLambdaRole],
-    }));
+    this.adminRole = new iam.Role(this, 'kubectlRole', {
+      assumedBy: new iam.CompositePrincipal(this.kubectlLambdaRole),
```

Comment:
For my understanding, why do we need to use `CompositePrincipal` here?

Reply:
Actually no need to use `CompositePrincipal` here. Will update.